### ISSUE TITLE
Support Run On Failure Keyword

### ISSUE THREAD

**Issue by robinmatz** [Altran-PT-GDC/Robot-Framework-Mainframe-3270-Library#42]
Hi there!

I have noticed that other GUI automation libraries tend to support a functionality that lets the automation engineer define what to do on keyword failure.

With `SeleniumLibrary`, `Browser`, and `AppiumLibrary`, to name a few, the default behavior is to take a screenshot of the application when a failure occurs. All three also support to change this default behavior to allow running another keyword, or none, by setting an option on library import or during test execution.

Right now I am using this workaround, which consist of taking a screenshot no matter what:
```
*** Test Cases ***
 Test Pass
     Open Connection    abc.com    23
     Write    some text
     Page Should Contain String    some text
     [Teardown]    Take Screenshot
     
Test Fail
     Open Connection    abc.com    23
     Write    some text
     Page Should Contain String    other text
     [Teardown]    Take Screenshot
```

Since most test cases do pass (and there is no requirement to save every screenshot), this results in a large number html files which are of no use.

Therefore, it would be great to have
1) An import parameter that defaults to `Take Screenshot` as the run on failure keyword. If set to ` None', no keyword should be run on failure.

2) A keyword like `Register run on failure keyword`, that can be set during test execution, to change the `run_on_failure` option temporarily.

Feel free to reach out for ideas and discussion!

**Comment by samuelpcabral** [comment]
Hi @robinmatz I undestand, but why not use inside the Teardown the BuildIn keyword: "Run Keyword If Test Failed" then you only take the screenshot if the test fails.

**Comment by robinmatz** [comment]
Hi @samuelpcabral thanks for your quick reply. 

I have tried the workaround you mentioned before. There are unfortunately several cases where this approach does not work. These are also stated in the official documentation: `This keyword can only be used in a test teardown. Trying to use it anywhere else results in an error.` [1](https://robotframework.org/robotframework/latest/libraries/BuiltIn.html#Run%20Keyword%20If%20Test%20Failed).

I took some time to look into how other libraries implemented their "run_on_failure" functionality. They tend to have extend their main library class from `DynamicCore` (from PythonLibCorePythonLibCore [2](https://github.com/robotframework/PythonLibCore)).

DynamicCore contains a run_keyword function that is overridden in the main library. There they  try to run a given keyword, catch an exception if it occurred. Then run the `run_on_failure`keyword and raise the error.

I tried to implement something similar for this libraries, and it seems to do the trick. I will open a PR for you to checkout the changes later.
